### PR TITLE
chore: upgrade ng from 11 to 12

### PR DESCRIPTION
**Pull request by itssharmasandeep**
Due to some problem with https://github.com/hypertrace/hyperdash-angular/pull/297
This is the new one with addressed comments

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,14 +17,14 @@
...
-    "@angular/platform-browser-dynamic": "^11.1.2",
-    "@angular/router": "^11.1.2",
+    "@angular/animations": "^12.2.1",
+    "@angular/common": "^12.2.1",
+    "@angular/compiler": "^12.2.1",
+    "@angular/core": "^12.2.1",
+    "@angular/forms": "^12.2.1",
+    "@angular/platform-browser": "^12.2.1",
+    "@angular/platform-browser-dynamic": "^12.2.1",
+    "@angular/router": "^12.2.1",
```
Comment (anandtiwary) [inline package.json:27]: what about zone.js? does it need to be upgraded?

File: src/environments/environment.ts [hunk 2]
```diff
@@ -14,4 +14,4 @@ export const environment = {
  * This import should be commented out in production mode because it will have a negative impact
  * on performance if an error is thrown.
  */
-// import 'zone.js/dist/zone-error';  // Included with Angular CLI.
+// import 'zone.js/plugins/zone-error';  // Included with Angular CLI.
```
Comment (anandtiwary) [inline src/environments/environment.ts:17]: I see this change here but no change in zone.js dependency. Is this right?

**Comment by anandtiwary** [comment]
@itssharmasandeep quick question: What is peer dependency in npm?

File: package.json (hunk 1 above)
Comment (itssharmasandeep) [reply package.json:27]: No, I think we're already on at latest one https://www.npmjs.com/package/zone.js?activeTab=versions

File: src/environments/environment.ts (hunk 2 above)
Comment (itssharmasandeep) [reply src/environments/environment.ts:17]: Yeah, answered above.

File: package.json [hunk 3]
```diff
@@ -3,12 +3,12 @@
   "version": "0.0.0",
   "scripts": {
     "ng": "ng",
-    "start": "ng serve",
-    "build": "ng build",
+    "start": "NG_PERSISTENT_BUILD_CACHE=1 ng serve",
+    "build": "NG_PERSISTENT_BUILD_CACHE-1 ng build",
```
Comment (aaron-steinfeld) [inline package.json:7]:
```suggestion
    "build": "NG_PERSISTENT_BUILD_CACHE=1 ng build",
```

File: src/environments/environment.ts (hunk 2 above)
Comment (aaron-steinfeld) [reply src/environments/environment.ts:17]: above you answered that zone.js doesn't need to be upgraded - but I think anand is asking: then why would this change? was it previously wrong?

It looks like the zone docs agree that it was:
https://github.com/angular/angular/tree/master/packages/zone.js#bundles

**Review by aaron-steinfeld — APPROVED** [review]

**Comment by anandtiwary** [comment]
We need to upgrade peer dependency before merging this pr. i have shared reading material with sandeep on slack

**Comment by aaron-steinfeld** [comment]
> We need to upgrade peer dependency before merging this pr. i have shared reading material with sandeep on slack

right - forgot this repo even had the separate project

**Review by anandtiwary — APPROVED** [review]

**Comment by itssharmasandeep** [comment]
taking `peerDependency` part separately

**Comment by aaron-steinfeld** [comment]
> taking peerDependency part separately

why? without the peer dependencies this isn't usable (so shouldn't have been merged). what am I missing?

**Comment by anandtiwary** [comment]
@aaron-steinfeld I wanted sandeep to try out and learn the entire process. Explained him few things, but a good hands on would help him learn the entire flow. So, I asked him to merge this PR and see what is breaking in ht-ui when he tries to add the new bundle.

**Pull request merged**